src/main/java/sora/util/Storage.java

```diff
@@ -0,0 +1,170 @@
+     */
+    public void save(List<Task> tasks) throws IOException {
+        // Reformat the list of tasks for storage
+        StringBuilder output = new StringBuilder();
+        tasks.forEach(task -> output.append(task.toString()).append("\n"));
+
+        // Overwrite the current save file
+        FileWriter fw = new FileWriter(filePath, false);
+        fw.write(output.toString());
+        fw.flush();
+        fw.close();
+    }
+
+    /**
+     * Loads data stored locally.
+     * If file does not exist, then a new file will be created.
+     *
+     * @return an array list of tasks containing any tasks that can be read from local storage
+     * @throws DataIntegrityException if the data in the save file is not in the correct format
+     * @throws IOException if the named file exists but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot be opened for any other reason
+     */
+    public ArrayList<Task> load() throws DataIntegrityException, IOException {
+        // List of tasks to be returned
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        // Initialize save file and create parent directory
+        File file = new File(filePath);
+
+        // Attempt to create the parent directory and the save file
+        if (file.createNewFile()) {
+            // File is created, exit function as nothing to read
+            return tasks;
+        }
+
+        // Attempt to load data
+        BufferedReader br = new BufferedReader(new FileReader(filePath));
+        String input;
+
+        // Read the rest of data and add to list of tasks
+        while ((input = br.readLine()) != null) {
+            // Add task to list but do not show a confirmation msg
+            tasks.add(newTask(input));
+        }
+
+        // Close reader
+        br.close();
+
+        return tasks;
+    }
+
+    private Task newTask(String command) throws DataIntegrityException {
+        Task newTask;
+
+        // Obtain relevant info based on type of task
+        switch (command.charAt(1)) {
+        case 'T': // todo, [T][ ] join sports club
+            newTask = new Todo(command.substring(7));
+            break;
+        case 'D': // deadline, [D][ ] return book (by: Jun 6 2021, 5:12 PM)
+            newTask = newDeadline(command);
+            break;
+        case 'E': // event, [E][ ] project meeting (at: Aug 6 2021, 2:00 PM - 6:00 PM)
+            newTask = newEvent(command);
+            break;
+        default: // gg someone messed with the save file
+            throw new DataIntegrityException();
+        }
+
+        // Check if task is alr done
+        if (command.charAt(4) == 'X') {
+            newTask.markAsDone();
+        }
+
+        return newTask;
+    }
+
+    private Task newEvent(String command) throws DataIntegrityException {
+        // [E][ ] project meeting (at: Aug 6 2021, 2:00 PM - 6:00 PM)
+
+        int index = command.lastIndexOf(" (at: ");
```

Comment:
Perhaps you can abstract this part & take in the prepositions "by" & "at" as parameters?

Reply:
This also, can you elaborate more? Do you mean I take `command.lastIndexOf("at")`?